TangemSdk/TangemSdk/Common/Card/Wallet.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Wallet.swift
+//  TangemSdk
+//
+//  Created by Alexander Osokin on 10.06.2021.
+//  Copyright © 2021 Tangem AG. All rights reserved.
+//
+
+import Foundation
+
+public extension Card {
+    /// Describing wallets created on card
+    struct Wallet: Codable {
+        /// Wallet's public key.
+        public let publicKey: Data
+        /// Elliptic curve used for all wallet key operations.
+        public let curve: EllipticCurve
+        /// Wallet's settings
+        public let settings: Settings
+        /// Total number of signed hashes returned by the wallet since its creation
+        /// COS 1.16+
+        public var totalSignedHashes: Int?
+        /// Remaining number of `Sign` operations before the wallet will stop signing any data.
+        /// - Note: This counter were deprecated for cards with COS 4.0 and higher
+        public var remainingSignatures: Int?
+        /// Index of the wallet in the card storage
+        internal let index: Int
+    }
+}
+
+public extension Card.Wallet {
+    struct Settings: Codable {
+        /// if true, erasing the wallet will be prohibited
+        public let isPermanent: Bool
+    }
+}
+
+extension Card.Wallet {
+    /// Status of the wallet
```

Comment:
```suggestion
    /// Status of the wallet. Only for cards with COS before v.4.0 
```
На самом деле можно еще туду написать, потому что непонятно для каких карт на самом деле. хочу пометить структуру. Потому что код читаешь и непонятно, что это уже депрекейтед.

Reply:
тоже самое и скард статусом